quickstart-node/src/services/monday.service.ts

```diff
@@ -0,0 +1,53 @@
+import { ApiClient, ChangeColumnValueOpMutation } from '@mondaydotcomorg/api';
+import { NotFound } from 'http-errors';
+import { GetItemsInGroupQuery, GetItemsInGroupQueryVariables, ItemWithColumnValuesFragment } from 'generated/graphql';
+import { getItemsInGroup } from 'queries.graphql';
+
+const UPDATED_MONDAY_API_VERSION = '2024-10';
+
+export class MondayService {
+  private constructor() {}
+
+  static async changeColumnValue(
+    token: string,
+    value: any,
+    itemId: string,
+    boardId: string,
+    columnId: string,
+  ): Promise<ChangeColumnValueOpMutation> {
+    const mondayApi = new ApiClient(token, UPDATED_MONDAY_API_VERSION);
+    return mondayApi.operations.changeColumnValueOp({
+      boardId,
+      itemId,
+      columnId,
+      value: value,
+    });
+  }
+
+  static async getLastItemInGroup(
+    token: string,
+    boardId: string,
+    groupId: string,
+  ): Promise<ItemWithColumnValuesFragment> {
+    const mondayApi = new ApiClient(token, UPDATED_MONDAY_API_VERSION);
+    const getItemsInGroupQueryVariables: GetItemsInGroupQueryVariables = {
+      groupId,
+      boardId,
+    };
+    const boards = await mondayApi.query<GetItemsInGroupQuery>(getItemsInGroup, getItemsInGroupQueryVariables);
+
+    if (boards?.boards?.length > 0) {
+      if (boards.boards[0].groups?.length > 0) {
+        if (boards.boards[0].groups[0].items_page?.items.length > 0) {
+          return boards.boards[0].groups[0].items_page?.items[0];
+        } else {
+          throw NotFound(`There are no items in board id: ${boardId} in group id: ${groupId}`);
+        }
+      } else {
+        throw NotFound(`Group id: ${groupId} in boardId: ${boardId} was not found`);
+      }
+    } else {
+      throw NotFound(`Board id: ${boardId} not found`);
+    }
+  }
```

Comment:
I think this code got a little bit too complicated to read 
we just need something like this 
```
const firstItem = boards?.boards?.[0]?.groups?.[0]?.items_page?.items?.[0];

if (firstItem) {
  return firstItem;
}
```

Reply:
if we want to add all the errors and all we can do it singular
```
const boards = await mondayApi.query<GetItemsInGroupQuery>(getItemsInGroup, getItemsInGroupQueryVariables);

const firstItem = boards?.boards?.[0]?.groups?.[0]?.items_page?.items?.[0];

if (firstItem) {
  return firstItem;
}

if (!boards?.boards?.length) {
  throw NotFound(`Board id: ${boardId} not found`);
}

if (!boards.boards[0].groups?.length) {
  throw NotFound(`Group id: ${groupId} in boardId: ${boardId} was not found`);
}

throw NotFound(`There are no items in board id: ${boardId} in group id: ${groupId}`);

```

Reply:
But I dont think we need to write the arguments in the error its against our standards. it need to be single message with params 
for instance
```
throw NotFound(`There are no items in board`, boardId: 123);
```

Reply:
I agree this error with parameters is good, but it will require more custom error classes which means more complexity to the project. Are you sure this is the right path and it is super important?